### PR TITLE
Adds usage of bookmarks from the browser panel

### LINKED ISSUE #4201: [FEATURE][browser] Add bookmarks data items to the browser panel
Original commit: https://github.com/qgis/QGIS/commit/b5b869b4f452f82af9a7d92d2d41e0f86468e876 by nirvn

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #4208: [FEATURE] Support drag and drop copying of bookmarks in browser, or
drag and drop to move bookmarks between groups
Original commit: https://github.com/qgis/QGIS/commit/87099bf558650a9fe0979de2d3040d28181077a2 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #4209: [FEATURE] Allow deletion of bookmarks through browser
Original commit: https://github.com/qgis/QGIS/commit/ab026b05ee52adabeae0390b2f75afb987c4f348 by nyalldawson

And port to QgsDataItemGuiProvider

### LINKED ISSUE #4210: [FEATURE] Allow bookmark renames via browser
Original commit: https://github.com/qgis/QGIS/commit/a162a880db611a638e8053bdd5fa43a27d99c1da by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #4215: [FEATURE] Bookmark groups can be renamed in browser
Original commit: https://github.com/qgis/QGIS/commit/a426fdaea259d08348410d61ade5f5d6c698b177 by nyalldawson

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by SrNetoChan**
Goal: Describe Spatial bookmarks usage from the browser panel, organize information in a table

Ticket(s): Fixes #4201, Fixes #4208, Fixes #4209, Fixes #4210, Fixes #4215

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -803,21 +806,67 @@ Note that you can have multiple bookmarks with the same name.
...
+       bookmarks, then select :guilabel:`Import Spatial Bookmarks`.
+   * - **Edit bookmark**
+     - You can change a bookmark by changing the values in the columns. You can
+       edit the name, the group, the extent and if its stored in the project or
+       not.
+     - Right-click the desired bookmark and select :guilabel:`Edit Spatial
+       Bookmark...`.
+       The :guilabel:`Bookmark editor` will open, allowing you to redefine every
+       aspect of the bookmark as if you were creating it from the first time.
+   * - **Move bookmark**
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/general_tools.rst]: I wonder if "move" is the right term, here. I first thought it was about the bookmark extent. Is it not still about editing the bookmark?

File: source/docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -803,21 +806,67 @@ Note that you can have multiple bookmarks with the same name.
...
+     - Select the bookmark and click :guilabel:`Delete Bookmark`.
+       Confirm your choice.
+     - Right-click the bookmark and select :guilabel:`Delete Bookmark`. Confirm
+       your choice.
+   * - **Export bookmarks to XML**
+     - Click :guilabel:`Import/Export Bookmarks` and select :guilabel:`Export
+       bookmarks`.
+       All bookmarks (user or project) are saved in a xml file.
+     - Right-click the Spatial bookmarks entry, or one of its folders (user and
+       project) or subfolders (groups) to determine which bookmarks to export,
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/introduction/general_tools.rst]: So the "group" column is only for the bookmarks tree in the browser, right? I was wondering what it was about.

File: source/docs/user_manual/introduction/general_tools.rst [hunk 3]
```diff
@@ -803,21 +806,67 @@ Note that you can have multiple bookmarks with the same name.
...
+   :header-rows: 1
+   :widths: 20 40 40
+   :stub-columns: 0
+
+   * - Task
+     - Spatial Bookmarks panel
+     - Browser panel
+   * - **Zoom to a Bookmark**
+     - Select the desired bookmark and click
+       :guilabel:`Zoom To Bookmark` or double-click on it.
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply source/docs/user_manual/introduction/general_tools.rst]: Agreed. All the actions in the bookmark panel are icons (with tool tip) based and I think we should keep showing them. Easier for readers.

[the remark this answers is not in the document]
Comment (SrNetoChan) [reply source/docs/user_manual/introduction/general_tools.rst]: Will do

File: source/docs/user_manual/introduction/general_tools.rst (hunk 1 above)
Comment (SrNetoChan) [reply source/docs/user_manual/introduction/general_tools.rst]: I will integrate this in the above row

**Comment by SrNetoChan** [comment]
@DelazJ @havatv added all icons and removed the "move bookmark row" please review.

**Comment by havatv** [comment]
Excellent, @SrNetoChan! I think that @DelazJ would like icons to also be included in the substitutions.rst file (source/docs/documentation_guidelines/substitutions.rst)

I guess these are the ones:
.. |sharingExport| image:: /static/common/mActionSharingExport.png
   :width: 1.5em
.. |sharingImport| image:: /static/common/mActionSharingImport.png
   :width: 1.5em
.. |zoomToBookmark| image:: /static/common/mActionZoomToBookmark.png
   :width: 1.5em

Hopefully @Delazj can advise on where (in which table) to place them.
I can also do it once @Delazj has given his advice.

**Pull request merged**